src/command/install.rs

```diff
@@ -0,0 +1,79 @@
+use semver::VersionReq;
+
+use notion_core::serial::version::parse_requirements;
+use notion_core::session::{ActivityKind, Session};
+use notion_fail::Fallible;
+
+use Notion;
+use command::{Command, CommandName, Help};
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Args {
+    arg_package: String,
+    arg_version: String,
+}
+
+pub(crate) enum Install {
+    Help,
+    Node(VersionReq),
+    Yarn(VersionReq),
+    Other { name: String, version: VersionReq },
+}
+
+impl Command for Install {
+    type Args = Args;
+
+    const USAGE: &'static str = "
+Install a global package
+
+Usage:
+    notion install <package> <version>
+    notion install -h | --help
+
+Options:
+    -h, --help     Display this message
+";
+
+    fn help() -> Self {
+        Install::Help
+    }
+
+    fn parse(
+        _: Notion,
+        Args {
+            arg_package,
+            arg_version,
+        }: Args,
+    ) -> Fallible<Self> {
+        match &arg_package[..] {
+            "node" => Ok(Install::Node(parse_requirements(&arg_version)?)),
+            "yarn" => Ok(Install::Yarn(parse_requirements(&arg_version)?)),
+            ref package => {
+                Ok(Install::Other {
+                    name: package.to_string(),
+                    version: parse_requirements(&arg_version)?,
+                })
+            }
+        }
+    }
+
+    fn run(self, session: &mut Session) -> Fallible<bool> {
+        session.add_event_start(ActivityKind::Install);
+        match self {
+            Install::Help => {
+                Help::Command(CommandName::Install).run(session)?;
+            }
+            Install::Node(requirements) => {
+                session.set_default_node(&requirements)?;
```

Comment:
So the most recently version of `node` or `yarn` that has been manually installed becomes the default global version? I think that's fine, since we're trying to get away from depending on the global toolchain state. And if the user really needs to care about the global version they can `notion install node <version>` for now. Or maybe we need something like `notion use node <version> --global` for this case?

Reply:
I'm not sure I follow -- can you say more?

Reply:
(pasted in from chat)

ok, I think I'm just hung up on command syntax
so `notion install node x.x.x` will install that node version if it's not already installed
or just set that version to be the default if it's already installed
same for yarn or anything else
thinking about the use cases for this, it doesn't quite make sense to me
as a user, I `notion install node 10`
then some time later I `notion install node 8`
which becomes the default
after that I want to switch back to node 10 for whatever reason
so I would run `notion install node 10` again
even though it's already installed
I think the problem is that I know that node 10 is already installed, so why do I need to install it again
but maybe it comes back to the stateless model of notion
like, who cares what is already installed
running notion commands is declaring how I want my system (or project) to be configured
so running `notion install node 10` is saying "I want to run this system with node v10, I don't care how you do it, just make it happen"
is `install` the right verb for that?
I guess, I can't think of anything better haha